### PR TITLE
Feature dynamic gbt

### PR DESCRIPTION
**Pull request by ramacarlucho**

### CODE REVIEW

**Review by dario-ramos — DISMISSED** [review]

File: include/bitcoin/node/full_node.hpp [hunk 1]
```diff
@@ -164,6 +164,7 @@ class BCN_API full_node
     const uint32_t protocol_maximum_;
     const node::settings& node_settings_;
     const blockchain::settings& chain_settings_;
+    mutable std::mutex gbt_mutex_;
```
Comment (mdallago) [inline include/bitcoin/node/full_node.hpp]: move to blockchain

File: src/protocols/protocol_block_in.cpp [hunk 2]
```diff
@@ -789,6 +789,8 @@ void protocol_block_in::handle_store_block(const code& ec, block_const_ptr messa
     // Show that diplayed forks may be missing activations due to checkpoints.
     const auto checked = state->is_under_checkpoint() ? "*" : "";
 
+    chain_.remove_mined_txs_from_mempool(message);
```
Comment (mdallago) [inline src/protocols/protocol_block_in.cpp]: move to blockchain

File: src/protocols/protocol_transaction_in.cpp [hunk 3]
```diff
@@ -249,6 +249,8 @@ void protocol_transaction_in::handle_store_transaction(const code& ec,
         return;
     }
 
+
+    chain_.add_to_chosen_list(message);
```
Comment (mdallago) [inline src/protocols/protocol_transaction_in.cpp:253]: move to blockchain

**Review by dario-ramos — DISMISSED** [review]

**Review by hanchon — DISMISSED** [review]

File: include/bitcoin/node/full_node.hpp (hunk 1 above)
Comment (fpelliccioni) [inline include/bitcoin/node/full_node.hpp]: the mutex is not used here, have to be a member of blockchain

**Review by fpelliccioni — CHANGES_REQUESTED** [review]

**Review by fpelliccioni — APPROVED** [review]

File: src/protocols/protocol_block_in.cpp (hunk 2 above)
Comment (mdallago) [reply src/protocols/protocol_block_in.cpp]: pending

File: src/protocols/protocol_transaction_in.cpp (hunk 3 above)
Comment (mdallago) [reply src/protocols/protocol_transaction_in.cpp:253]: pending

**Review by mdallago — COMMENTED** [review]
pending items

**Review by mdallago — APPROVED** [review]

**Review by fpelliccioni — APPROVED** [review]
Needs a refactoring for the next version

**Pull request merged**